gitapp/lib/user_details/details_page.dart

```diff
@@ -1,37 +1,86 @@
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:gitapp/constants/constants.dart';
-import 'package:gitapp/user_details/user_details_model.dart';
 import 'package:gitapp/user_details/user_avatar_widget.dart';
+import 'package:gitapp/user_details/user_details_repository.dart';
 import 'package:gitapp/user_details/user_info_widget.dart';
 import 'package:gitapp/user_details/user_repo_model.dart';
+import 'package:gitapp/user_details/user_details_model.dart';
 import 'package:gitapp/user_details/user_repos_list_widget.dart';
 import 'package:gitapp/user_details/user_details_presenter.dart';
 
-class DetailScreen extends StatelessWidget {
+Future<List<UserRepo>> repositoriesList(int userId) async =>
+    await UserDetailsPresenter.userDetailsPresenter.getRepositoriesList(userId);
+Future<UserDetails> userDetails(int userId) async =>
+    await UserDetailsRepository.userDetailsRepository.fetchUserDetails(userId);
 
-  DetailScreen({required this.user});
-  final String user;
+class DetailScreen extends StatefulWidget {
+  DetailScreen({required this.userId});
+  final int userId;
 
   @override
-  Widget build(BuildContext context) {
+  _DetailScreenState createState() => _DetailScreenState();
+}
 
-    UserDetails mock = getMockedInfo(user);
-    List<UserRepo> repo = getMockedRepos(user);
+class _DetailScreenState extends State<DetailScreen> {
+  @override
+  void initState() {
+    super.initState();
+    userDetails(widget.userId);
+    repositoriesList(widget.userId);
+  }
 
+  @override
+  Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
         title: Text("User details"),
       ),
-      body: Column(
-        children: [
-          SizedBox(height: 20.0),
-          UserAvatarWidget(username: mock.username, avatarUrl: mock.avatarUrl),
-          UserInfoWidget(followersCount: mock.followersCount, repositoriesCount: mock.repositoriesCount, country: mock.country),
-          Divider(height: 35.0, color: yyellow, indent: 30.0, endIndent: 30.0),
-          UserReposWidget(listRepos: repo),
-        ]
+      body: SingleChildScrollView(
+        child: FutureBuilder(
+          future: Future.wait(
+              [userDetails(widget.userId), repositoriesList(widget.userId)]),
+          builder: (BuildContext context, AsyncSnapshot<dynamic> snapshot) {
+            if (snapshot.hasData) {
```

Comment:
Each if case can be moved to different widget. It would be easier to understand for someone.

Reply:
Extracted two of them, the latter is not up to be extracted because it has reference to an enclosing class method.